FutureNorthants/northants-design-system
# Add event list and event page

Components and page designs for the event list and event pages. 
https://github.com/FutureNorthants/northants-website/issues/1095 


## Components added
- EventDetails - layout for the date, time and location
- EventLink - layout for event used in the EventList. 
- RelatedEvents - A list of events that are related to another event. Uses the EventLink component. 
- EventList - layout for all events. This is a controlled component so the state is managed outside this component in the frontend. 

## Components changed
- The DropDownSelect has been updated so the state can either be managed by the component or as a controlled component so it can be managed outside the component. 

## Page mockups
- [Event Landing page example](https://deploy-preview-529--northants-design-system.netlify.app/?path=/story/page-examples-event-landing-page-example--event-landing-page) - shows the EventList component in situ. 
- [Event page example](https://deploy-preview-529--northants-design-system.netlify.app/?path=/story/page-examples-event-page-example--event-page) - uses existing components to present a possible layout for the event page. 

## Testing
- Checkout this branch and run `npm install` then `npm run test` to run the test script
- Run `npm run dev` and manually test the new components and pages. Ensure the styles look ok and there are no obvious errors. 
- Run `npm run build` to test the build script works as expected without any TypeScript issues.
